src/scitacean/transfer/link.py

```diff
@@ -0,0 +1,186 @@
+    """Connection for 'uploading' files with symlinks.
+
+    Should be created using
+    :meth:`scitacean.transfer.link.LinkFileTransfer.connect_for_upload`.
+
+    Is not actually implemented!
+    """
+
+    def __init__(self, *, source_folder: RemotePath) -> None:
+        self._source_folder = source_folder
+
+    @property
+    def source_folder(self) -> RemotePath:
+        """The source folder this connection uploads to."""
+        return self._source_folder
+
+    def remote_path(self, filename: Union[str, RemotePath]) -> RemotePath:
+        """Return the complete remote path for a given path."""
+        return self.source_folder / filename
+
+    def upload_files(self, *files: File) -> List[File]:
+        """Upload files to the remote folder."""
+        raise NotImplementedError()
+
+    def revert_upload(self, *files: File) -> None:
+        """Remove uploaded files from the remote folder."""
+        raise NotImplementedError()
+
+
+class LinkFileTransfer:
+    """Upload / download files by creating symlinks.
+
+    This file transfer does not actually upload or download files.
+    Instead, it requires that the 'remote' file system is directly
+    accessible from the 'local' file system.
+    It creates symlinks in the local download folder to the
+    'remote' files.
+
+    Note
+    ----
+    A note on terminology:
+    In Scitacean, 'remote' refers to the file server where the data files
+    are stored that belong to SciCat datasets.
+    In contrast, 'local' refers to the file system of the machine that
+    runs the Python process.
+    The two filesystems can be the same.
+    However, Scitacean maintains a strict separation between the two and
+    uses 'downloaders' and 'uploaders' to transfer between them even if that
+    transfer is a simple symlink.
+
+    See also the documentation of :class:`scitacean.File`.
+
+    Warning
+    -------
+    This file transfer does not work on Windows.
+    This is due to :class:`RemotePath` not supporting backslashes as
+    path separators.
+    In practice, this should not be a problem because SciCat file storage
+    should never be a Windows server.
+
+    Warning
+    -------
+    This file transfer cannot upload files.
+    Instead, consider copying or moving the files to the SciCat source folder
+    or writing the files there directly from your workflow.
+
+    Attempting to upload files will raise ``NotImplementedError``.
+
+    Examples
+    --------
+    Given a dataset with ``source_folder="/dataset/source"`` and a file with path
+    ``"file1.dat"``, this
+
+    .. code-block:: python
+
+        client = Client.from_token(
+            url="...",
+            token="...",
+            file_transfer=LinkFileTransfer()
+        )
+        ds = client.get_dataset(pid="...")
+        ds = client.download_files(ds, target="/downloads")
+
+    creates the following symlink:
+
+    .. code-block::
+
+        /downloads/file1.dat -> /dataset/source/file1.dat
```

Comment:
Should we be making symlinks or should we be making copies?
I guess when you are using a sftp file transfer, you make a copy. And then you are allowed to manipulate/modify the file you get as you wish.

If we make symlinks to read-only files, we won't then be able to do the same.
This would imply that using a different file-transfer would give you different results, even if the file you are targetting is the same. Does it seem inconsistent?